doit/cmd_base.py

```diff
@@ -578,6 +578,10 @@ def execute(self, params, args):
         if not legacy_loader:
             self.task_list = self.loader.load_tasks(cmd=self, pos_args=args)
 
+        # Add task options from config, if present
+        for task in self.task_list:
+            task.cfg_defaults = self.config.get(task.name, {})
```

Comment:
It bothers me a bit that we might be calling `config.get` for every task when rarely it will contain a value...

Reply:
It is roughly 30% faster to do an `if task.name in self.config: ...`. What the actual execution time benefit would be depends on how many tasks are defined.

Benchmark:

https://gist.github.com/rbdixon/797d0478f9716c4123e401a1187552e1

```
Number of tasks: 1326
Speedup -30.40%
Baseline duration: 0.25ms
Alternative duration: 0.18ms
```

Might be hard for a user to tell the difference in execution time either way.